pkg/plugins/golang/options.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package golang
+
+import (
+	"fmt"
+	"path"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kubebuilder/v2/pkg/internal/validation"
+	"sigs.k8s.io/kubebuilder/v2/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/v2/pkg/model/resource"
+)
+
+const (
+	versionPattern  = "^v\\d+(?:alpha\\d+|beta\\d+)?$"
+	groupPresent    = "group flag present but empty"
+	versionPresent  = "version flag present but empty"
+	kindPresent     = "kind flag present but empty"
+	versionRequired = "version cannot be empty"
+	kindRequired    = "kind cannot be empty"
+)
+
+var (
+	versionRegex = regexp.MustCompile(versionPattern)
+
+	coreGroups = map[string]string{
+		"admission":             "k8s.io",
+		"admissionregistration": "k8s.io",
+		"apps":                  "",
+		"auditregistration":     "k8s.io",
+		"apiextensions":         "k8s.io",
+		"authentication":        "k8s.io",
+		"authorization":         "k8s.io",
+		"autoscaling":           "",
+		"batch":                 "",
+		"certificates":          "k8s.io",
+		"coordination":          "k8s.io",
+		"core":                  "",
+		"events":                "k8s.io",
+		"extensions":            "",
+		"imagepolicy":           "k8s.io",
+		"networking":            "k8s.io",
+		"node":                  "k8s.io",
+		"metrics":               "k8s.io",
+		"policy":                "",
+		"rbac.authorization":    "k8s.io",
+		"scheduling":            "k8s.io",
+		"setting":               "k8s.io",
+		"storage":               "k8s.io",
+	}
+)
+
+// Options contains the information required to build a new resource.Resource.
+type Options struct {
+	// Group is the resource's group. Does not contain the domain.
+	Group string
+	// Domain is the resource's domain.
+	Domain string
+	// Version is the resource's version.
+	Version string
+	// Kind is the resource's kind.
+	Kind string
+
+	// Plural is the resource's kind plural form.
+	// Optional
+	Plural string
+
+	// CRDVersion is the CustomResourceDefinition API version that will be used for the resource.
+	CRDVersion string
+	// WebhookVersion is the {Validating,Mutating}WebhookConfiguration API version that will be used for the resource.
+	WebhookVersion string
+
+	// Namespaced is true if the resource should be namespaced.
+	Namespaced bool
+
+	// Flags that define which parts should be scaffolded
+	DoAPI        bool
+	DoController bool
+	DoDefaulting bool
+	DoValidation bool
+	DoConversion bool
```

Comment:
IMO these attrs brings confusion. They have not the name of the flags used in the ux and they are not the model ones. Then, you are here creating a third representation for values that came from the flags and are stored in the model. 

We should ONLY create new attrs/options for what we need to manipulate and we do not have a direct representation on the model. Otherwise, we bring a lot of unnecessary complexities.

Reply:
These fields are not represented in the model anywhere. They are only represented in the webhook scaffolder. Previously they didn't have the `Do` prefixes, I added the prefixes to keep the same pattern as in the API scaffolder.